elyra/metadata/metadata_app.py

```diff
@@ -473,6 +473,96 @@ def start(self):
+
+    def start(self):
+        super().start()  # process options
+
+        directory = self.directory_option.value
+
+        src_directory = os.path.join(directory, self.schemaspace)
+
+        try:
+            json_files = [f for f in os.listdir(src_directory) if f.endswith('.json')]
+        except OSError as e:
+            print(f"Unable to reach the '{src_directory}' directory: {e.strerror}: '{e.filename}'")
+            self.exit(1)
+
+        if len(json_files) == 0:
+            print(f"No instances for import found in the '{src_directory}' directory")
+            return
+
+        instance_count_imported = 0
+        instance_count_not_imported = 0
+        metadata_file = None
+
+        for file in json_files:
+            filepath = os.path.join(src_directory, file)
+            try:
+                with open(filepath) as f:
+                    metadata_file = json.loads(f.read())
+                f.close()
+            except OSError as e:
+                print(f"Unable to reach the '{file}' file: {e.strerror}: '{e.filename}'")
+                instance_count_not_imported += 1
+                continue
+
+            name = os.path.splitext(file)[0]
+            try:
+                schema_name = metadata_file['schema_name']
+                display_name = metadata_file['display_name']
+                metadata = metadata_file['metadata']
+            except KeyError as e:
+                print(f"Could not find '{e.args[0]}' key in the import file '{filepath}'")
+                instance_count_not_imported += 1
+                continue
+
+            new_instance = None
+
+            try:
+                if self.replace_flag.value:  # if replacing, fetch the instance so it can be updated
+                    updated_instance = self.metadata_manager.get(name)
+                    updated_instance.schema_name = schema_name
+                    if display_name:
+                        updated_instance.display_name = display_name
+                    if name:
+                        updated_instance.name = name
+                    updated_instance.metadata.update(metadata)
+                    new_instance = self.metadata_manager.update(name, updated_instance)
+                else:  # create a new instance
+                    instance = Metadata(schema_name=schema_name, name=name,
+                                        display_name=display_name, metadata=metadata)
+                    new_instance = self.metadata_manager.create(name, instance)
+            except Exception:
+                pass
+
+            if new_instance:
+                instance_count_imported += 1
+            else:
+                instance_count_not_imported += 1
+
+        print(f"\nImported {instance_count_imported} " + ("instance" if instance_count_imported == 1 else "instances") +
+              ((f"\n{instance_count_not_imported} " + ("instance" if instance_count_not_imported == 1
```

Comment:
Let's split this into 2 prints.  We might want to add a third with something like:
`The following files could not be imported:  foo.json, bar.json` (per previous comment)

Reply:
@kevin-bates Apologies for the delay - I made all the suggested updates.

Currently whenever a file is not imported, I am printing out the error msg, and later on I am printing out the names of all the files that weren't imported (As per your comment). I'm wondering if the two should be combined so that the output is something like:

Imported 6 instances
1 instance could not be imported

The following files could not be imported: 
|File|Reason|
|-|-|
|invalid.json|Validation failed for instance 'invalid' using the metadata-test schema with error: '//localhost:8081/' is not a 'uri'. |

Reply:
Hi @sandhya-nayak - no worries about the delay - it's a busy world out there!

I think your suggestion to postpone the details until after the summary is great.  There will be some "noise" that is logged by the server code (if the admin didn't redirect output to a file), but this suggestion makes good sense regardless.

Reply:
@kevin-bates updated - please review when you can. Thanks!